src/mbgl/util/tile_cover_impl.cpp

```diff
@@ -234,47 +235,48 @@ TileCoverImpl::TileCoverImpl(int32_t z, const Geometry<double>& geom, bool proje
     ToFeatureType toFeatureType;
     isClosed = apply_visitor(toFeatureType, geom) == FeatureType::Polygon;
 
-    ToEdgeTables toEdgeTables(z, project);
-    edgeTables = apply_visitor(toEdgeTables, geom);
+    ToEdgeTable toEdgeTable(z, project);
+    edgeTable = apply_visitor(toEdgeTable, geom);
     reset();
 }
 
 void TileCoverImpl::reset() {
-    currentEdgeTable = edgeTables.begin();
-    if (currentEdgeTable != edgeTables.end()) {
-        activeEdgeTable = currentEdgeTable->begin()->second;
-        currentY = currentEdgeTable->begin()->first;
-    }
+    activeEdgeTable = edgeTable.begin()->second;
+    currentRow = edgeTable.begin()->first;
 }
 
 bool TileCoverImpl::next() {
-    return currentEdgeTable != edgeTables.end() && activeEdgeTable.size() != 0 && currentY < maxY;
+    return activeEdgeTable.size() != 0 && currentRow < maxY;
 }
 
 bool TileCoverImpl::scanRow(ScanLine& scanCover) {
     if (!next()) { return false; }
 
-    auto xps = util::scan_row(currentY, activeEdgeTable);
-    auto p = xps.begin();
+    auto xps = util::scan_row(currentRow, activeEdgeTable);
     assert(isClosed ? xps.size() % 2 == 0 : true);
-    while (p != xps.end()) {
-        auto x_min = p->first;
-        // For closed geometries, consider odd even pairs of tile ranges
-        if (isClosed) { p++; }
-        auto x_max = p++->second;
-        scanCover(x_min, x_max, currentY);
+
+    auto x_min = xps[0].x0;
+    auto x_max = xps[0].x1;
+    int8_t nzRule = xps[0].winding ? 1 : -1;
```

Comment:
`nzRule` can overflow if the scan line intersects a ton of line strings, right? Maybe worth bumping `int8_t` to 16 or 32 just to be safe.